refinery/data_set_manager/utils.py

```diff
@@ -1192,3 +1194,94 @@ def get_solr_response_json(node_uuids):
         'data_set_manager'
     )
     return format_solr_response(solr_response)
+
+
+def update_existing_dataset_with_revised_investigation(existing_dataset_uuid,
+                                                       revised_investigation):
+    """
+    Update an existing DataSet's Investigation with a new Investigation as
+    long as some specific constraints are met.
+    - Said DataSet must be "clean" (No Analyses or Visualizations performed)
+    - All datafiles from the existing DataSet must be referenced in the new
+    Investigation
+    Any data files that were uploaded prior to this operation will be
+    reassociated and available from the new Investigation
+    :param revised_investigation: A newly created Investigation instance
+    :param existing_dataset_uuid: the UUID of an existing DataSet
+    """
+    existing_dataset = core.models.DataSet.objects.get(
+        uuid=existing_dataset_uuid
+    )
+    associate_datafiles_from_existing_dataset(
+        existing_dataset, revised_investigation
+    )
+    updated_dataset_title = revised_investigation.title
+    existing_dataset.update_investigation(
+        revised_investigation,
+        "Metadata Revision: for {}".format(updated_dataset_title)
+    )
+    data_set_manager.tasks.annotate_nodes(revised_investigation.uuid)
+    existing_dataset.set_title(updated_dataset_title)
+
+
+@transaction.atomic()
+def associate_datafiles_from_existing_dataset(existing_dataset,
```

Comment:
The long name is a code smell. This should simply be a method of the DataSet model.